pkg/instrumentation/kafka/kafka.go

```diff
@@ -0,0 +1,169 @@
+package kafka
+
+import (
+	"context"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"math"
+)
+
+type (
+	KafkaClient interface {
+		Produce(ctx context.Context, r *kgo.Record, promise func(*kgo.Record, error))
+		ProduceSync(ctx context.Context, rs ...*kgo.Record) kgo.ProduceResults
+		PollRecords(ctx context.Context, num int) kgo.Fetches
+		Flush(ctx context.Context) error
+		Close()
+	}
+
+	Client struct {
+		KafkaClient
+		cfg  *config
+		prev ddtrace.Span
+	}
+
+	FetchesRecordIter struct {
+		*kgo.FetchesRecordIter
+		ctx    context.Context
+		client *Client
+	}
+)
+
+// NewClient calls kgo.NewClient and wraps the resulting Client.
+func NewClient(conf []kgo.Opt, opts ...Option) (*Client, error) {
+	c, err := kgo.NewClient(conf...)
+	if err != nil {
+		return nil, err
+	}
+
+	return WrapClient(c, opts...), nil
+}
+
+// WrapClient wraps kgo.Client so that produced and consumed messages are traced.
+func WrapClient(c KafkaClient, opts ...Option) *Client {
+	return &Client{
+		KafkaClient: c,
+		cfg:         newConfig(opts...),
+	}
+}
+
+func (c *Client) startProducerSpan(ctx context.Context, msg *kgo.Record) ddtrace.Span {
+	opts := []tracer.StartSpanOption{
+		tracer.ServiceName(c.cfg.producerServiceName),
+		tracer.ResourceName("Produce Topic " + msg.Topic),
```

Comment:
@fotos All Kafka client wrapper in datadog library ([kafka-go](https://github.com/DataDog/dd-trace-go/blob/d57ebc3f216cec67a9baa3f5f57dd44f3978df0b/contrib/segmentio/kafka.go.v0/kafka.go#L116) and [confluent-kafka-go](https://github.com/DataDog/dd-trace-go/blob/d57ebc3f216cec67a9baa3f5f57dd44f3978df0b/contrib/confluentinc/confluent-kafka-go/kafka/kafka.go#L201)) use the same naming conventions. I'd rather keep it consistent here, esp in case we want to move this part to the `datadog` lib in the future.